### PR TITLE
Unify rescue from Pundit::NotAuthorizedError

### PR DESCRIPTION
**Pull request by hennevogel**
So you can use `Pundit::NotAuthorizedError.reason` also in the API.
We need that because we have many actions that do not only check
permissions but also processability.

### CODE REVIEW

**Comment by dmarcoux** [comment]
The failing specs need to be fixed.

File: src/api/app/controllers/concerns/rescue_authorization_handler.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    def authorization_errorcode(exception)
+      "#{action_for_exception(exception)}_#{ActiveSupport::Inflector.underscore(exception.record.class.to_s)}_not_authorized"
+    end
+
+    def authorization_message(exception)
+      case exception.reason
+      when :anonymous_user
+        'Please login to access the resource'
+      else
+        "Sorry, you are not authorized to #{action_for_exception(exception)} this #{ActiveSupport::Inflector.underscore(exception.record.class.to_s).humanize}."
```
Comment (dmarcoux) [inline src/api/app/controllers/concerns/rescue_authorization_handler.rb:47]: To prevent having a random capitalized word in the middle of a sentence, we could do this:

```suggestion
        "Sorry, you are not authorized to #{action_for_exception(exception)} this #{ActiveSupport::Inflector.underscore(exception.record.class.to_s).humanize(capitalize: false)}."
```

Comment (hennevogel) [reply src/api/app/controllers/concerns/rescue_authorization_handler.rb]: Pushed

File: src/api/app/controllers/webui/groups_controller.rb [hunk 2]
```diff
@@ -4,20 +4,20 @@ class Webui::GroupsController < Webui::WebuiController
   after_action :verify_authorized, except: [:show, :autocomplete]
 
   def index
-    authorize Group, :index?
+    authorize Group.new, :index?
```
Comment (dmarcoux) [inline src/api/app/controllers/webui/groups_controller.rb:7]: This is a bit weird, we're not going to create groups in the `index` view. Is this change really needed?

Comment (hennevogel) [reply src/api/app/controllers/webui/groups_controller.rb:7]:
```
[4] pry(main)> Group.new.class.name
=> "Group"
[5] pry(main)> Group.class.name
=> "Class"
```

**Comment by hennevogel** [comment]
coverage job has succeeded see [cicle](https://app.circleci.com/pipelines/github/openSUSE/open-build-service/7319/workflows/11704265-4b3a-4cd5-85cc-33c52dc1f087/jobs/89554)

**Review by dmarcoux — APPROVED** [review]

**Comment by hennevogel** [comment]
coveralls hangs. codecov is fine. merging!
